### ISSUE TITLE
bug: `get_waku_v2_relay_v1_messages ` return an array has payload instead of expected hex string

### ISSUE THREAD

**Issue by fryorcraken** [waku-org/nwaku#1139]
### Problem

`get_waku_v2_relay_v1_messages ` returns an array of decimal number in payload instead of the expected hex string defined in https://rfc.vac.dev/spec/16/#get_waku_v2_relay_v1_messages

### Impact

Minor defect, just unexpected as an implementer.

### To reproduce

1. Ensure the node receives some messages over relay

2. JSON RPC call:
````
 get_waku_v2_relay_v1_messages [ '/waku/2/default-waku/proto' ]
```

JSON RPC Response:

```
{
  "jsonrpc": "2.0",
  "id": 1,
  "result": [
    {
      "payload": [
        76,
        105,
        103,
        104,
        116,
        32,
        80,
        117,
        115,
        104,
        32,
        119,
        111,
        114,
        107,
        115,
        33
      ],
      "contentTopic": "/test/1/waku-light-push/utf8",
      "version": 0,
      "timestamp": 1662821422291000000,
      "proof": {
        ...
      }
    }
  ]
}
```

### Expected behavior

`payload` is a hex string.

### nwaku version/commit hash

`v0.11`

**Comment by fryorcraken** [comment]
Odd, the crypto API returns hex:

```

**Comment by fryorcraken** [comment]
@richard-ramos I have witnessed the same behaviour on go-waku

**Comment by richard-ramos** [comment]
@fryorcraken in go-waku I added this behavior to match nwaku's. Otherwise Go would have encoded the byte array as base64 (which is what it's used in REST API)

**Comment by rymnc** [comment]
Upon some investigation, 

- `get_waku_v2_relay_v1_messages` does not serialize the `WakuMessage` for the response
- `get_waku_v2_private_v1_asymmetric_messages` makes use of this [function](https://github.com/status-im/nwaku/blob/01cd201f5890f4870d18daef899c66d165efb2cd/waku/v2/node/waku_payload.nim#L40) which applies the conversion 

- The REST API has a few helper functions for this - https://github.com/status-im/nwaku/blob/7f23bdf29fc60d89a3ec1d2f759e8bc037122833/waku/v2/node/rest/relay/relay_api.nim#L156-L175

**Comment by jm-clius** [comment]
@fryorcraken @richard-ramos is it worth getting a fix in nwaku (and parity in the other clients)? Although the JSON-RPC API is going to be deprecated, we have to maintain it while still being used.

**Comment by fryorcraken** [comment]
@notdao have you noticed the issue? Do you use this API?

**Comment by fryorcraken** [comment]
IMHO, it'd be cleaner but now that I deal with it, I am not fussed:

https://github.com/waku-org/js-waku/blob/5bab85f93f22f853f0d2d07d134b8c5d0d4a93f1/src/test_utils/nwaku.ts#L316

@dao any thoughts on the matter as you use the JSON RPC API?

**Comment by dao** [comment]
> IMHO, it'd be cleaner but now that I deal with it, I am not fussed:
> 
> https://github.com/waku-org/js-waku/blob/5bab85f93f22f853f0d2d07d134b8c5d0d4a93f1/src/test_utils/nwaku.ts#L316
> 
> @dao any thoughts on the matter as you use the JSON RPC API?

I vote for consistency - easy enough for us to make this change once

**Comment by jm-clius** [comment]
@rymnc this is not priority. Given your involvement with RLN, should we unassign you here and move this item to a backlog?

**Comment by rymnc** [comment]
Unassigned, although I would suggest to leave it in `Todo` since there are quite a few consumers of the json-rpc api, as @LNSD suggested. Will leave it up to you to decide though!

**Comment by LNSD** [comment]
As @richard-ramos commented above, the efficient way to serialize the messages' payload is to use base64 encoding. 

I am assigning this issue to me, as I think it is feasible to fix this as part of v0.15.0.
